### DISCUSSION TITLE
WKT file for the stadium geometry

### DISCUSSION THREAD

**Discussion by nchuly** [PedestrianDynamics/jupedsim#1431] [Q&A]
Hi, thanks for the great work! I am currently working on a project that involves simulating pedestrian dynamics in a stadium-like environment, and I came across [this link](https://www.youtube.com/watch?v=jHZ-uVQ076c&list=PLYiH62IDwy5c1MSKt-9bvtmkgzYDVXBKv&index=19) which seems highly relevant to my work.Could you kindly let me know if it is possible to obtain the WKT file for the stadium geometry referenced in the provided link? It would greatly help in accurately modeling the environment for my simulation.

**Comment by chraibi** [comment]
Hi,
thanks for your feedback and glad you like it.
Oh, you picked a geometry that was created 10 years ago!!  ^_^

I found in a deep dark corner in my hard desk some xml files, that I could also convert to wkt files. 
Hope that works for you. 
[jul.zip](https://github.com/user-attachments/files/18101686/jul.zip)

**Comment by nchuly** [comment]
it works, thanks!!^_^

**Comment by nchuly** [comment]
Sorry to bother you again, but I have another question. I noticed that the WKT file you provided contains only `LINESTRING` geometries. I was wondering how such data could be used effectively in Jupedsim. Specifically, I would like to know:

+ Can LINESTRING geometries be directly utilized in Jupedsim for defining boundaries, obstacles, or walkable areas?
+ If not, is there a recommended way to convert these LINESTRING geometries into POLYGON format for compatibility with Jupedsim?
I appreciate your insights and guidance on this matter.

**Comment by chraibi** [comment]
According to [this page](https://www.jupedsim.org/stable/concepts/geometry.html), it seems you can't use the WKT files as they are.

That's why I shared the parser code and the XML files with you—so you can explore and figure it out. I think the parser needs to be updated to convert the subrooms from the XML files into Polygons instead of just collections of LineStrings.

Want to give it a shot?

**Comment by nchuly** [comment]
got it! thanks!!^_^

**Comment by Ozaq** [comment]
I took the liberty to move this into a discussion as this is not an issue with JuPedSim itself.

**Comment by chraibi** [comment]
Hi @nchuly did you make some progress? I hope so. In that case it would be nice if you could share your files. 
Maybe we can add new notebooks using the new geometry.
